src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -0,0 +1,459 @@
+            //Starting with number of threads as 10
+            int numThreads = 10;
+            int trackComponent = 0;
+            final UnirestInstance ui = UnirestFactory.getUnirestInstance();
+            while (trackComponent < paginatedList.size()) {
+                for (int i = 0; i < numThreads; i++) {
+                    final List<Component> temp = new ArrayList<>();
+                    int k = 0;
+                    while (trackComponent < paginatedList.size() && k < numThreads) {
+                        temp.add(paginatedList.get(trackComponent));
+                        k += 1;
+                        trackComponent += 1;
+                    }
+
+                    Runnable analysisUtil = () -> {
+                        for (final Component component : temp) {
+                            final String snykUrl = API_BASE_URL + parsePurlToSnykUrlParam(component.getPurl()) + API_ENDPOINT;
+                            if (component.getPurl() != null) {
+                                if (!isCacheCurrent(Vulnerability.Source.SNYK, API_BASE_URL, component.getPurl().toString())) {
+                                    try {
+                                        final GetRequest request = ui.get(snykUrl)
+                                                .header(HttpHeaders.AUTHORIZATION, this.apiToken);
+                                        final HttpResponse<JsonNode> jsonResponse = request.asJson();
+                                        if (jsonResponse.getStatus() == 200 || jsonResponse.getStatus() == 404) {
+                                            handle(component, jsonResponse.getBody().getObject(), jsonResponse.getStatus(), jsonResponse.getStatusText());
+                                        } else {
+                                            handleUnexpectedHttpResponse(LOGGER, API_BASE_URL, jsonResponse.getStatus(), jsonResponse.getStatusText());
+                                        }
+                                    } catch (UnirestException e) {
+                                        handleRequestException(LOGGER, e);
+                                    }
+                                } else {
+                                    applyAnalysisFromCache(Vulnerability.Source.SNYK, API_BASE_URL, component.getPurl().toString(), component, getAnalyzerIdentity(), vulnerabilityAnalysisLevel);
+                                }
+                            }
+                        }
+                    };
+                    Instant startThread = Instant.now();
+                    executor.execute(analysisUtil);
+                    Instant endThread = Instant.now();
+                    duration += Duration.between(startThread, endThread).toMillis();
+                }
+            }
+
+            paginatedComponents.nextPage();
+        }
+    }
+
+    public void handle(Component component, JSONObject object, int responseCode, String statusText) {
```

Comment:
As @mehab already pointed out, there's a lot of stuff going on in this class, and this method in particular. I'd suggest moving the parsing logic to dedicated classes in the `org.dependencytrack.parser.snyk` package.

Reply:
working on this one

Reply:
done